MaterialLib/MPL/Properties/PermeabilityMohrCoulombFailureIndexModel.h

```diff
@@ -0,0 +1,127 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on June 4, 2020, 10:13 AM
+ */
+
+#pragma once
+
+#include "MaterialLib/MPL/Property.h"
+#include "MaterialLib/MPL/VariableType.h"
+
+namespace ParameterLib
+{
+struct CoordinateSystem;
+template <typename T>
+struct Parameter;
+}  // namespace ParameterLib
+
+namespace MaterialPropertyLib
+{
+/**
+ * \brief A failure index dependent permeability model \cite wangetalPerm2020
+ *
+ * \f[ \mathbf{k} =
+ *      \mathbf{k}_0+ H(f-1) k_\text{r} \mathrm{e}^{b f}\mathbf{I}\f]
+ *
+ * where
+ *   \f$\mathbf{k}_0\f$ is the intrinsic permeability.
+ *  of the undamaged material,
+ * \f$H\f$ is the Heaviside step function, \f$f\f$ is the failure index,
+ * \f$k_\text{r}\f$ is a reference permeability,
+ * \f$b\f$ is a fitting parameter.
+ * \f$k_\text{r}\f$ and \f$b\f$ can be calibrated by experimental data.
+ *
+ * The failure index \f$f\f$ is calculated from  the Mohr Coulomb failure
+ * criterion comparing an acting shear stress for the shear dominated failure.
+ * The tensile failure is governed by an input parameter of
+ * tensile_strength_parameter .
+ *
+ *  The Mohr Coulomb failure
+ * criterion \cite labuz2012mohr takes
+ * the form
+ *   \f[\tau(\sigma)=c-\sigma \mathrm{tan} \phi\f]
+ *   with \f$\tau\f$ the shear stress, \f$c\f$ the cohesion, \f$\sigma\f$ the
+ * tensile stress, and \f$\phi\f$ the internal friction angle.
+ *
+ *  The failure index of the Mohr Coulomb model is calculated by
+ *   \f[
+ *         f_{MC}=\frac{|\tau_m| }{\cos(\phi)\tau(\sigma_m)}
+ *   \f]
+ *   with
+ *   \f$\tau_m=(\sigma_3-\sigma_1)/2\f$
+ *   and \f$\sigma_m=(\sigma_1+\sigma_3)/2\f$,
+ *  where \f$\sigma_1\f$ and \f$\sigma_3\f$ are the minimum and maximum shear
+ * stress, respectively.
+ *
+ * The tensile failure index is calculated by
+ *  \f[
+ *    f_{t} = \sigma_m / \sigma^t_{max}
+ * \f]
+ * with, \f$0 < \sigma^t_{max} < c \tan(\phi) \f$, a parameter of tensile
+ * strength for the cutting of the apex of the Mohr Coulomb model.
+ *
+ * The tensile stress status is determined by a condition of \f$\sigma_m>
+ * \sigma^t_{max}\f$. The failure index is then calculated by
+ * \f[
+ *   f =
+ *  \begin{cases}
+ *    f_{MC}, & \sigma_{m} \leq \sigma^t_{max}\\
+ *    max(f_{MC}, f_t), & \sigma_{m} > \sigma^t_{max}\\
+ *  \end{cases}
+ * \f]
+ *
+ *  The computed permeability components are restricted with an upper bound,
+ *  i.e. \f$\mathbf{k}:=k_{ij} < k_{max}\f$.
```

Comment:
But only for some specific case, not in all cases. And also before rotation.

Reply:
It for the computed value of k=k0+H(f−1) k_r exp(bI). We do not need to write that restriction in the source code for H(f-1) =0 (f<1) for k=k0. Therefore it is a general expression of the restriction of the permeability of the model. The rotation (only needed for symmetrical k tensor) is a mapping between the two systems, and the restriction have to be applied in the original system, i.e. the local system.

Reply:
I was too fast to close it. I agree on the first part. But the documentation should mention, that the restriction is done before rotation and not after. The result is different, *e.g.* if k = Id and k_max is 0.5. Restrict and rotate yields different value than rotate and restrict.

Reply:
The restriction is only for the model. The rotation is computational stuff. Anyway, added a sentence in the class documentation:
```
 * If \f$\mathbf{k}_0\f$ is orthogonal, i.e input two or three numbers
 * for the diagonal entries of \f$\mathbf{k}_0\f$, a coordinate system rotation
 * is possible if it is needed.
```